Index: mediawiki/extensions/CategoryTree/modules/ext.categoryTree/ext.categoryTree.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mediawiki/extensions/CategoryTree/modules/ext.categoryTree/ext.categoryTree.js b/mediawiki/extensions/CategoryTree/modules/ext.categoryTree/ext.categoryTree.js
--- a/mediawiki/extensions/CategoryTree/modules/ext.categoryTree/ext.categoryTree.js	(revision d287b0fcaae07df1d22514cb9ad67744b4df05c1)
+++ b/mediawiki/extensions/CategoryTree/modules/ext.categoryTree/ext.categoryTree.js	(date 1675431299409)
@@ -205,6 +205,8 @@
 	$( function () {
 		// Attach click handler for sidebar
 		attachHandler( $( '#p-categorytree-portlet' ) );
+
+		attachHandler( $('#mw-data-after-content') );
 	} );
 
 }() );
