Index: mediawiki/extensions/VisualEditor/modules/ve-mw/ui/inspectors/ve.ui.MWAlienExtensionInspector.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mediawiki/extensions/VisualEditor/modules/ve-mw/ui/inspectors/ve.ui.MWAlienExtensionInspector.js b/mediawiki/extensions/VisualEditor/modules/ve-mw/ui/inspectors/ve.ui.MWAlienExtensionInspector.js
--- a/mediawiki/extensions/VisualEditor/modules/ve-mw/ui/inspectors/ve.ui.MWAlienExtensionInspector.js	(revision 1a2ec45b6ccacb87caf49058f93dd105def2afd4)
+++ b/mediawiki/extensions/VisualEditor/modules/ve-mw/ui/inspectors/ve.ui.MWAlienExtensionInspector.js	(date 1660290113317)
@@ -60,6 +60,10 @@
 			var key, attributeInput, field,
 				attributes = this.selectedNode.getAttribute( 'mw' ).attrs;

+			// Patch: KK to extend tag edit panel
+			if (ve.ui.MWAlienExtensionInspectorExtension) {
+				ve.ui.MWAlienExtensionInspectorExtension.extend(this);
+			}
 			if ( attributes && !ve.isEmptyObject( attributes ) ) {
 				for ( key in attributes ) {
 					attributeInput = new OO.ui.TextInputWidget( {