Index: mediawiki/extensions/EnhancedRetrieval/src/FacetedSearch/FSSolrSMWDB.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mediawiki/extensions/EnhancedRetrieval/src/FacetedSearch/FSSolrSMWDB.php b/mediawiki/extensions/EnhancedRetrieval/src/FacetedSearch/FSSolrSMWDB.php
--- a/mediawiki/extensions/EnhancedRetrieval/src/FacetedSearch/FSSolrSMWDB.php	(revision 931855bd6f602cd1c46854f65be1fe04f49cd469)
+++ b/mediawiki/extensions/EnhancedRetrieval/src/FacetedSearch/FSSolrSMWDB.php	(date 1712237967749)
@@ -116,6 +116,10 @@
         $pageDbKey  = $pageTitle->getDBkey();
         $text = $rawText ?? $this->getText( $wikiPage, $doc, $messages );
 
+        $extText = '';
+        \Hooks::run('FS_ExtendSearchFulltext', [& $extText]);
+        $text .= $extText;
+
         $doc['id'] = $pageID;
         $doc['smwh_namespace_id'] = $pageNamespace;
         $doc['smwh_title'] = $pageDbKey;
