Index: mediawiki/resources/src/mediawiki.searchSuggest/searchSuggest.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mediawiki/resources/src/mediawiki.searchSuggest/searchSuggest.js b/mediawiki/resources/src/mediawiki.searchSuggest/searchSuggest.js
--- a/mediawiki/resources/src/mediawiki.searchSuggest/searchSuggest.js	(revision a2581929e8a86ac8124cc0636c4322f2bb61f236)
+++ b/mediawiki/resources/src/mediawiki.searchSuggest/searchSuggest.js	(date 1669982429586)
@@ -235,7 +235,7 @@
 
 					api = api || new mw.Api();
 
-					$.data( node, 'request', mw.searchSuggest.request( api, query, response, maxRows ) );
+					$.data( node, 'request', mw.searchSuggest.request( api, query, response, maxRows, mw.config.get('fsgNamespacesForSearchField') || undefined ) );
 				},
 				cancel: function () {
 					var node = this[ 0 ],
