Index: mediawiki/extensions/PageForms/libs/ext.pf.select2.combobox.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mediawiki/extensions/PageForms/libs/ext.pf.select2.combobox.js b/mediawiki/extensions/PageForms/libs/ext.pf.select2.combobox.js
--- a/mediawiki/extensions/PageForms/libs/ext.pf.select2.combobox.js	(revision faf2dd17463db26e0eaafdf02e525f147a5ae53e)
+++ b/mediawiki/extensions/PageForms/libs/ext.pf.select2.combobox.js	(date 1667576086086)
@@ -81,7 +81,7 @@
 			if ( term === undefined ) {
 				term = "";
 			}
-			return pf.select2.base.prototype.textHighlight( result.id, term );
+			return pf.select2.base.prototype.textHighlight( result.label ? result.label : result.id, term );
 		}
 		opts.language.searching = function() {
 			return mw.msg( "pf-select2-searching" );
