Index: mediawiki/extensions/PageForms/libs/PF_ComboBoxInput.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mediawiki/extensions/PageForms/libs/PF_ComboBoxInput.js b/mediawiki/extensions/PageForms/libs/PF_ComboBoxInput.js
--- a/mediawiki/extensions/PageForms/libs/PF_ComboBoxInput.js	(revision 85f4113555081430abc672eef2e4ef0d8552ec1c)
+++ b/mediawiki/extensions/PageForms/libs/PF_ComboBoxInput.js	(date 1697127400616)
@@ -161,7 +161,7 @@
                                     self.itemFound = true;
                                 }
                                 values.push({
-                                    data: Data[i].title, label: self.highlightText(Data[i].title)
+                                    data: Data[i].title, label: self.highlightText(Data[i].label ? Data[i].label : Data[i].title)
                                 })
                             }
                         }
