Index: mediawiki/extensions/VisualEditor/lib/ve/src/ui/ve.ui.PositionedTargetToolbar.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mediawiki/extensions/VisualEditor/lib/ve/src/ui/ve.ui.PositionedTargetToolbar.js b/mediawiki/extensions/VisualEditor/lib/ve/src/ui/ve.ui.PositionedTargetToolbar.js
--- a/mediawiki/extensions/VisualEditor/lib/ve/src/ui/ve.ui.PositionedTargetToolbar.js	(revision 0cd5464eb4e6885b2282560e35d75b007c2cf4d2)
+++ b/mediawiki/extensions/VisualEditor/lib/ve/src/ui/ve.ui.PositionedTargetToolbar.js	(date 1661427885018)
@@ -100,7 +100,8 @@
 	if ( this.floating ) {
 		this.$bar.css( {
 			left: this.elementOffset.left,
-			right: this.elementOffset.right
+			right: this.elementOffset.right,
+			top: '50px'
 		} );
 	}
 
@@ -149,7 +150,8 @@
 			.addClass( 've-ui-toolbar-floating' );
 		this.$bar.css( {
 			left: this.elementOffset.left,
-			right: this.elementOffset.right
+			right: this.elementOffset.right,
+			top: '50px'
 		} );
 		this.floating = true;
 		this.emit( 'resize' );
@@ -166,7 +168,7 @@
 		this.$element
 			.css( 'height', '' )
 			.removeClass( 've-ui-toolbar-floating' );
-		this.$bar.css( { left: '', right: '' } );
+		this.$bar.css( { left: '', right: '', top: '' } );
 		this.floating = false;
 		this.emit( 'resize' );
 		this.onViewportResize();
