Index: mediawiki/extensions/PageForms/libs/PageForms.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mediawiki/extensions/PageForms/libs/PageForms.js b/mediawiki/extensions/PageForms/libs/PageForms.js
--- a/mediawiki/extensions/PageForms/libs/PageForms.js	(revision 91ed23245f8627973ea43548cfd5f2a4954ada19)
+++ b/mediawiki/extensions/PageForms/libs/PageForms.js	(date 1706799040365)
@@ -2006,8 +2006,10 @@
 	$('div.mw-collapsed a.mw-collapsible-text').click();
 });
 
-$('.pfSendBack').click( function() {
-	window.history.back();
+$('.pfSendBack').click( function(e) {
+	if (window.top == window.self) {
+		window.history.back();
+	}
 });
 
 }( jQuery, mediaWiki ) );
Index: mediawiki/extensions/PageForms/libs/PF_popupform.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mediawiki/extensions/PageForms/libs/PF_popupform.js b/mediawiki/extensions/PageForms/libs/PF_popupform.js
--- a/mediawiki/extensions/PageForms/libs/PF_popupform.js	(revision 91ed23245f8627973ea43548cfd5f2a4954ada19)
+++ b/mediawiki/extensions/PageForms/libs/PF_popupform.js	(date 1706796813236)
@@ -746,6 +746,7 @@
 		// catch 'Cancel'-Link (and other 'back'-links) and close frame instead of going back
 		var backlinks = allLinks.filter('a[href="javascript:history.go(-1);"]');
 		backlinks.click(handleCloseFrame);
+		$('.pfSendBack', window.frames[0].document).off('click').click(handleCloseFrame);
 
 		// promote any other links to open in main window, prevent nested browsing
 		allLinks
